include/ola/base/Macro.h

```diff
@@ -63,6 +63,30 @@
 #endif  // __GNUC__
 
 /**
+ * @def OLA_FALLTROUGH
+ * @brief Mark switch cases as falltrough when required
+ *
+ * @examplepara
+ *   @code
+ *   switch (cond) {
+ *     case 'foo':
+ *       doFoo();
+ *       //after foo falltrough to bar because $REASON
+ *       OLA_FALLTROUGH
+ *     case 'bar':
+ *   @endcode
+ */
+#ifdef __GNUC__
+#if __GNUC__ >= 7
+#define OLA_FALLTROUGH __attribute__ ((fallthrough));
```

Comment:
From the issue, we want to use [[fallthrough]]; when the compiler has C++17 support, regardless of which compiler it is, but that may be a future change I imagine. Perhaps add it as a todo if it doesn't work now (which it shouldn't as we're forcing something older to bypass other issues.